torch/_functorch/aot_autograd.py

```diff
@@ -3913,11 +3951,25 @@ def backward(ctx, *flat_args):
 Expected grad_output types: {str(CompiledFunction.metadata.output_types)}
 Got grad_output types: {str(grad_output_types)}"""
 
-            del contiguous_args
-
             # TODO: figure out how to refactor the backward properly so I can use aot_dispatch_subclass_wrapper() here.
+            tangents_start_idx = len(all_args) - num_flat_bw_args_with_grads - len(rng_args)
+            tangents_end_idx = len(all_args) - len(rng_args)
+
             if CompiledFunction.maybe_subclass_metadata is not None:
+                # Get the number of tangents after unwrapping
+                len_tangents = len(unwrap_tensor_subclasses(
+                    all_args[tangents_start_idx: tangents_end_idx], is_joint_structure=False
+                ))
                 all_args = unwrap_tensor_subclasses(all_args, is_joint_structure=False)
+                tangents_start_idx = len(all_args) - len_tangents - len(rng_args)
+                tangents_end_idx = tangents_start_idx + len_tangents
+
+            # Make the tangents contiguous. Note that we must do this after subclass desugaring
+            # because inputs to inductor have to be contiguous
+            all_args = [
+                t.contiguous() if isinstance(t, torch.Tensor) and tangents_start_idx <= i < tangents_end_idx else t
```

Comment:
hmm, is the `isinstance(t, torch.Tensor)` check necessary here? I might be wrong but I thought that all tangents have to be tensors (we won't e.g. have symints here, since symints will be directly saved in the ctx).

Reply:
Ah I wasn't sure because this was preexisting and I was worried about Nones from undefined gradients, but I think we need to materialize them today, so I think you're right.